### ISSUE TITLE
Generating F1 Score, recall, and precision with Yolo11-cls

### ISSUE THREAD

**Issue by NotPhinx** [ultralytics/ultralytics#17817]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I want to be able to calculate or access these metrics for my classification model. I have tried everything but I can't find how to access the information. I tried setting plots=true to display the charts for these metrics, but it also did not work. I was just wondering if there was something I was doing wrong or if this was even possible with a classification model considering they are not metrics returned when validating.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @NotPhinx, thank you for your interest in Ultralytics 🚀! We recommend checking the [Docs](https://docs.ultralytics.com) for detailed information on Ultralytics usage, where many common questions and examples are addressed, including [classification-specific details](https://docs.ultralytics.com/tasks/classification/).

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us debug it. This will allow us to better understand the issue you're encountering.

If this is a custom ❓ Question, such as calculating metrics for classification models, please provide additional details about your experiment setup, including:

- The exact command or code you used
- Relevant sections of your `cfg` or setup file (if applicable)
- Logs or outputs, especially from training or validation phases

For insights into training and validating classification models, also ensure you’re following our [Best Training Practices](https://docs.ultralytics.com/guides/model-training-tips/).

## Join the Community

Get involved and share knowledge with other users:
- Real-time discussions on [Discord](https://ultralytics.com/discord) 🎧
- In-depth threads on [Discourse](https://community.ultralytics.com)
- Engaging conversations on our [Subreddit](https://reddit.com/r/ultralytics)

## Upgrade

Ensure you're using the latest version of the `ultralytics` package and its [dependencies](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) under a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/). Upgrade the package by running:

```bash
pip install -U ultralytics
```

Upgrading often resolves issues, as the latest version may include bug fixes and new features.

## Environments

To simplify experiments, YOLO can run in various environments with all dependencies pre-installed, including:

- **Notebooks**: Run YOLO with free GPU support: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM: See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI: See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**: Use the official <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a> and see our [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/)

## Status

To ensure systems are functioning as expected:
<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If green, all tests for YOLO modes (including classification) are passing on macOS, Windows, and Ubuntu as part of our [Continuous Integration](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule).

We hope this helps! This is an automated response to expedite assistance, but an Ultralytics engineer will review your question and respond soon 🙂

**Comment by Y-T-G** [comment]
For classification models, these metrics are not generated.

**Comment by NotPhinx** [comment]
> For classification models, these metrics are not generated.

Do you happen to know how I would be able to calculate these on my own? Or are these not really important for a classification model

**Comment by Y-T-G** [comment]
You can calculate them using `scikit-learn`.

https://scikit-learn.org/1.5/modules/generated/sklearn.metrics.classification_report.html

`y_true` would list of labels. `y_pred` would the be class output by the model.

**Comment by NotPhinx** [comment]
> You can calculate them using `scikit-learn`.
> 
> https://scikit-learn.org/1.5/modules/generated/sklearn.metrics.classification_report.html
> 
> `y_true` would list of labels. `y_pred` would the be class output by the model.

how would you access the class output on the model that's the problem I ran into when I tried to use this before so I just thought that you could not easily access this.

**Comment by Y-T-G** [comment]
You just create two lists, `y_true` and `y_pred`. For every image, you append to the `y_true` list the actual label and to the `y_pred` list you append `result.probs.top1`.

Then you just pass them to the scikit-learn function.
